src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,50 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(_headers is null)
+                {
+                    return new List<IHttpCookie>();
+                }
+
+                // this produces either an empty list (no cookie) or a list with a single KeyValuePair
+                // in the format of ("Cookie", "cookie_1=value;cookie_2=value...")
+                var cookieList = _headers.ToLookup(item => item.Key)["Cookie"].ToList();
+
+                if (cookieList.Count > 0)
+                {
+                    // cookieList should have only one KeyValuePair, where the Value is all of the cookies concatenated
+                    // together as a string (ex. "Cookie_1=value;Cookie_2=value;Cookie_3=value")
+                    return cookieList.ConvertAll(new Converter<KeyValuePair<string, IEnumerable<string>>, IReadOnlyCollection<IHttpCookie>>(CookieStringsToHttpCookie)).First();
+                }
+
+                return new List<IHttpCookie>();
+               
+            });
+        }
+
+        public static IReadOnlyCollection<IHttpCookie> CookieStringsToHttpCookie(KeyValuePair<string, IEnumerable<string>> cookieStrings)
+        {
+            // cookieStrings.Value comes in the format "Cookie_1=value;Cookie_2=value;Cookie_3=value"
+            var separateCookies = cookieStrings.Value.First().ToString().Split(";");
+
+            List<IHttpCookie> httpCookiesList = new List<IHttpCookie>();
```

Comment:
There is an overload which takes the capacity of the list. If you know how many items you are going to add to the list before hand, you can use that overload. This will prevent some array allocation & copying (_List internally allocates a small array first and when you add items more than that capacity, List will create a new array, double the size of current capacity and copy the elements from first array to the new array. This is how pretty much all dynamic collections auto grows._)

`new List<IHttpCookie>(separateCookies.Length);`